File: src/main/java/io/sch/historyscan/domain/logging/spi/Logger.java
package io.sch.historyscan.domain.logging.spi;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

@HexagonalArchitectureSPI
public interface Logger {

    void info(String message);

    void error(Exception ex);

    void error(String message);

    void debug(String message);

    void error(String message, Exception exception);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/logging/spi/Logger.kt
package io.sch.historyscan.domain.logging.spi

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI

@HexagonalArchitectureSPI
interface Logger {
    fun info(message: String)

    fun error(ex: Exception)

    fun error(message: String)

    fun debug(message: String)

    fun error(message: String, exception: Exception)
}
